[Bugfix] Updated the participant list UIs to match the latest UX requirement and Android for consistency

kevinyulu opened the pull request:
## Purpose
Updated the participant list UIs to match the latest UX requirement and Android for consistency
* Muted icon: use the unfilled style as indicated in the spec and on Android
* Updated display name local participant postfix from "(You)" to "(me)"
* Updated participant avatar style conditionally based on unnamed participant
* Updated text colour of unnamed participant name

![Simulator Screen Shot - iPhone 12 Pro - 2022-01-27 at 00 19 19](https://user-images.githubusercontent.com/90668345/151322402-49bdda4f-88cb-4d60-8a16-5feeff2df57a.png)


## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x] Other... Please describe: UX update
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code
<!-- Add steps to run the tests suite and/or manually test -->
* Join a call (using group or Teams) (with name filled, or without name filled)
* Launch participant drawer
* Follow **What to Check** section


## What to Check
Verify that the following are valid
In participant drawer: 
* Check if the muted icon style is the unfilled (regular) style as indicated in the spec and on Android 
* Check the display name local participant postfix is "(me)"
* Check if the participant avatar style (abbreviation) is conditionally based on the unnamed participant
* Check if the text colour of unnamed participant name is (#6E6E6E, light) and (#919191, dark)
* Check if the colour of the unfilled (regular) style **mute** icon is (#ACACAC, light) and (#919191, dark)

## Other Information
<!-- Add any other helpful information that may be needed here. -->

Commit:
Updated the participant list and resolved UX differences between Android & iOS

* Muted icon: use the unfilled style as indicated in the spec and on Android
* Updated display name local participant postfix from "(You)" to "(me)"
* Updated participant avatar style conditionally based on unnamed participant
* Updated text colour of unnamed participant name

JoshuaLai commented on AzureCommunicationUI/AzureCommunicationUI/Presentation/FluentUI/Wrapper/CompositeParticipantsListCell.swift:
```diff
@@ -10,12 +10,13 @@ class CompositeParticipantsListCell: TableViewCell {
         let isNameEmpty = displayName.trimmingCharacters(in: .whitespaces).isEmpty
         let avatar = MSFAvatar(style: .accent, size: .medium)
-        avatar.state.primaryText = displayName
+        let isDisplayNameUnnamed = displayName.contains(StringConstants.defaultEmptyName)
```
I am a little confused why we are doing this extra `let` here? Doesn't the `isNameEmpty` tell us the exact same thing?  
My only concern here is in the very odd edge case where a user actually types in `Unnamed participant` as a display name. 
The check below I think can just be 
```
avatar.state.primaryText = isNameEmpty? nil : displayName
```
And we can remove the extra `isDisplayNameUnnamed` in the or condition below.

kevinyulu replied:
I think when a user leaves the displayName empty, the local user viewModel updates the name to "Unnamed Participant" and this "Unnamed Participant" is passed here in the `setup`. Thus the `isNameEmpty` will return `false`. Then this `isDisplayNameUnnamed` was used to check whether the name contains "Unnamed Participant". 

> My only concern here is in the very odd edge case where a user actually types in Unnamed participant as a display name.

yeah there is chance of this. I'll check

Commit:
Made app have a default name for remote and local participant

Also added a new attribute indicating if the participant name is empty in ParticipantListCellViewModel.

Commit:
start to use outlined style for anonymous user, accent otherwise

AmyL219 commented on AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingViewComponent/ParticipantsListCellViewModel.swift:
```diff
@@ -7,21 +7,31 @@ import Foundation
+            self.displayName = "\(StringConstants.defaultEmptyName)"
+            self.isParticipantNameEmpty = true
+        } else {
+            self.displayName = participantInfoModel.displayName
```
Can't let  `displayName =  participantInfoModel.displayName` in the viewModel. 
For specific logic how to handle displaying empty name, put it inside the participantsListCell?
Which could remove `isParticipantNameEmpty` property here

AmyL219 replied:
Feel like using `displayName.isEmpty` as the single source of truth is easier for understanding and maintainrance

JoshuaLai commented on AzureCommunicationUI/AzureCommunicationUI/Presentation/FluentUI/Wrapper/CompositeParticipantsListCell.swift:
```diff
@@ -7,15 +7,26 @@ import Foundation
+    ///   - accessoryType: accessory type
+    func setup(displayName: String,
+               isMuted: Bool,
+               partipantHasEmptyName: Bool,
```
2 things. 
Spelling here:
```suggestion
               participantHasEmptyName: Bool,
```
And if we are now passing in a boolean to represent this I don't think we need the 
`let isNameEmpty = displayName.trimmingCharacters(in: .whitespaces).isEmpty` so we should remove that.

Commit:
Moved the displayName empty check logic into the CompositeParticipantListCell

Also made the CompositeAvatar style conditionally based on display name emptiness

kevinyulu committed:
Merge branch 'develop/beta' into bug/participantListUXUpdate

kevinyulu committed:
Merge branch 'develop/beta' into bug/participantListUXUpdate

AmyL219 commented:
Sorry, just remember of the ChangeLog update. 
Could you update the changelog to include this bug fix in your next PR?
